src/discordtipbot/discordbot.cr

```diff
@@ -184,12 +185,13 @@ class DiscordBot
     channel(msg).guild_id.as(UInt64)
   end
 
-  private def amount(msg : Discord::Message, string)
+  private def amount(msg : Discord::Message, string) : BigDecimal?
     if string == "all"
       amount = @tip.get_balance(msg.author.id)
-    else
-      if m = /(?<amount>^[0-9,\.]+)/.match(string)
-        amount = m["amount"].try &.to_f64
+    elsif m = /(?<amount>^[0-9,\.]+)/.match(string)
+      begin
+        BigDecimal.new(m["amount"]).round(8)
+      rescue
```

Comment:
Can we do this without blindly rescuing any possible exception?

Reply:
It's the easiest/safest way of doing it IMO
Or if you have a better regex as suggestion I'd also be up for that.

Reply:
I'm terrible with regex, so not really. At least rescue `InvalidBigDecimalException`, though.